test/client.test.js

```diff
@@ -1946,6 +1946,47 @@ describe('ACC Client', function () {
             var extAccount = await query.executeQuery();
             expect(extAccount).toEqual({ extAccount: [] });
         });
+
+        it("Expired session callBack", async () => {
+            let onSessionExpiration = jest.fn();
+            const connectionParameters = sdk.ConnectionParameters.ofBearerToken("http://acc-sdk:8080", 
+                                                    "$token$", {onSessionExpiration: onSessionExpiration});
+            const client = await sdk.init(connectionParameters);
+            client._transport = jest.fn();
+            client._transport.mockReturnValueOnce(Mock.BEARER_LOGON_RESPONSE);
+            await client.logon();
+            client._transport.mockReturnValueOnce(Mock.GET_XTK_QUERY_SCHEMA_RESPONSE);
+            var queryDef = {
+                "schema": "nms:extAccount",
+                "operation": "select",
+                "select": {
+                    "node": [
+                        { "expr": "@id" },
+                        { "expr": "@name" }
+                    ]
+                }
+            };
+
+            client._transport.mockReturnValueOnce(Promise.resolve(`XSV-350008 Session has expired or is invalid. Please reconnect.`));
+
+            var query = client.NLWS.xtkQueryDef.create(queryDef);
+            await expect(query.executeQuery()).rejects.toMatchObject({ errorCode: "XSV-350008" });
+            expect(onSessionExpiration.mock.calls.length).toBe(1);
+            
+            client._transport.mockReturnValueOnce(Promise.resolve(`<?xml version='1.0'?>
```

Comment:
I think it would be useful to illustrate how to log on again after we have a new bearer token

Reply:
I added an example login gain with the same credential. I think we need to  modify the client in order to support the use a new bearer token with the same client.